lib/osml/data_intake/di_dataplane.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2023-2024 Amazon.com, Inc. or its affiliates.
+ */
+
+import { Duration, RemovalPolicy, Size } from "aws-cdk-lib";
+import { ISecurityGroup, SecurityGroup } from "aws-cdk-lib/aws-ec2";
+import { IRole } from "aws-cdk-lib/aws-iam";
+import { DockerImageCode, DockerImageFunction } from "aws-cdk-lib/aws-lambda";
+import { Bucket } from "aws-cdk-lib/aws-s3";
+import { Topic } from "aws-cdk-lib/aws-sns";
+import { LambdaSubscription } from "aws-cdk-lib/aws-sns-subscriptions";
+import { Construct } from "constructs";
+
+import { OSMLAccount } from "../osml_account";
+import { OSMLBucket } from "../osml_bucket";
+import { OSMLVpc } from "../osml_vpc";
+import { DILambdaRole } from "./roles/di_lambda_role";
+
+/**
+ * Configuration class for DIDataplane Construct.
+ */
+export class DIDataplaneConfig {
+  /**
+   * Constructor for DIDataplane Construct.
+   * @param {string} LAMBDA_ROLE_NAME - The name of the DI Lambda execution role.
+   * @param {string} LAMBDA_FUNCTION_NAME - The name of the Lambda for the Image Handler.
+   * @param {number} LAMBDA_MEMORY_SIZE - The memory in Mb to give the lambda runtime.
+   * @param {number} LAMBDA_STORAGE_SIZE - The size of the storage to assign lambda runtime in GB.
+   * @param {string} SNS_INPUT_TOPIC_NAME - The name to give the input SNS topic.
+   * @param {string} SNS_OUTPUT_TOPIC_NAME - The name to give the output SNS topic.
+   * @param {string} S3_INPUT_BUCKET_NAME - The name to give the input bucket.
+   * @param {string} S3_OUTPUT_BUCKET_NAME - The name to give the output bucket.
+   */
+  constructor(
+    public LAMBDA_ROLE_NAME: string = "DILambdaRole",
+    public LAMBDA_FUNCTION_NAME: string = "DILambda",
+    public LAMBDA_MEMORY_SIZE: number = 1024,
```

Comment:
May want to increase these memory and storage size limits. Not hard to find an image that is larger than 1GB.

Reply:
Fixed